docs/src/environments.md

```diff
@@ -112,3 +112,146 @@ Otherwise, it will resolve the latest versions of the dependencies compatible wi
     ```bash
     $ julia --project=. myscript.jl
     ```
+
+
+## Temporary environment
+
+It is not uncommon to test some package without installing it into your usual environment. For instance,
+when you try to write a bug report with a consistent way to reproduce, or you want to try out a very
+old/new package version that is incompatible with your current environment.
+
+```julia-repl
+(@v1.6) pkg> activate --temp # require Julia 1.5 or later
+  Activating new environment at `/var/folders/34/km3mmt5930gc4pzq1d08jvjw0000gn/T/jl_a31egx/Project.toml`
+
+(jl_a31egx) pkg> add Example
+    Updating registry at `~/.julia/registries/General`
+   Resolving package versions...
+    Updating `/private/var/folders/34/km3mmt5930gc4pzq1d08jvjw0000gn/T/jl_a31egx/Project.toml`
+  [7876af07] + Example v0.5.3
+    Updating `/private/var/folders/34/km3mmt5930gc4pzq1d08jvjw0000gn/T/jl_a31egx/Manifest.toml`
+  [7876af07] + Example v0.5.3
+```
+
+The temporary directory will be removed when you exit Julia so you don't need to worry this operation
+breaks your common workflow.
+
+Note that this still reuses the contents in the global depot path in `~/.julia`. Thus you might still hit
```

Comment:
I felt this started to get way too long. I have almost never needed to change the depot path because things in it are content tracked so it should never " break your common workflow.". We have to try keep the text a bit short so readers don't get fatigued.